### PR TITLE
CouchDB 2.0 compatibility + flattening of relational pouch data section.

### PR DESCRIPTION
**Pull request by afinne**
Example configuration:

```
require 'couch_tap'

changes 'http://admin:admin@localhost:5984/fieldops_m3data' do
  database 'postgres://localhost:5432/afinne'
  document 'type' => 'errorSymptom' do
    table :error_symptoms do
      column :short_description, :shortDescription
    end
  end

  document 'type' => 'technician' do
    table :technicians do
      column :name, :name
      column :warehouse, :warehouse
    end
  end
end

changes 'http://admin:admin@localhost:5984/fieldops_location' do
  database 'postgres://localhost:5432/afinne'
  document 'type' => 'location' do
    table :locations do
      column :customer_number, :customerNumber
      column :customer_name, :customerName
    end
  end
end
```

### CODE REVIEW

File: lib/couch_tap/changes.rb [hunk 1]
```diff
@@ -127,7 +127,13 @@ def process_row(row)
...
 
     def fetch_document(id)
-      source.get(id)
+      doc = source.get(id)
+      transform_document(doc)
+    end
+
+    def transform_document(doc)
+      return doc unless doc['data']
+      doc.merge(doc.delete('data'))
```
Comment (perlun) [inline lib/couch_tap/changes.rb:136]: Very hardwired for our use case, but fine apart from that.

File: lib/couch_tap/changes.rb [hunk 2]
```diff
@@ -141,19 +147,26 @@ def find_or_create_sequence_number
...
     def update_sequence(seq)
-      database[:couch_sequence].where(:name => source.name).update(:seq => seq)
+      data = {
+        name: source.name,
+        seq: seq,
+        updated_at: DateTime.now
+      }
+      database[:couch_sequence]
+        .insert_conflict(target: :name, update: data)
+        .insert(data.merge(created_at: data[:updated_at]))
```
Comment (perlun) [inline lib/couch_tap/changes.rb:157]: Using upsert: good.

File: lib/couch_tap/changes.rb [hunk 3]
```diff
@@ -141,19 +147,26 @@ def find_or_create_sequence_number
...
       self.seq = seq
     end
 
     def create_sequence_table
       database.create_table :couch_sequence do
         String :name, :primary_key => true
-        Bignum :seq, :default => 0
+        String :seq, :default => 0
```
Comment (perlun) [inline lib/couch_tap/changes.rb:164]: Why not a Bignum here?

**Review by perlun — COMMENTED** [review]
A question that would be nice to have an answer to first.

File: lib/couch_tap/changes.rb (hunk 3 above)
Comment (afinne) [reply lib/couch_tap/changes.rb:164]: In CouchDb 1.6, seq was a number, in CouchDb 2.0 seq is a convoluted hashwannabethingmagigmegastring: `42-g1AAAACbeJzLYWBgYMpgTmEQTM4vTc5ISXLIyU9OzMnILy7JAUklMiTV____PyuDOVEoFyjAbphkbGFuaIJNAx5j8liAJEMDkPoPNU0CbFqKZbKZmYElNn1ZAFfbMG8`

Comment (perlun) [reply lib/couch_tap/changes.rb:164]: Ah. Makes perfect sense!

**Review by perlun — APPROVED** [review]
Fine change, thanks.

**Pull request merged**